File: Backend/SEPT_Backend/src/main/java/com/sept_group6/sept_backend/model/Patient.java
package com.sept_group6.sept_backend.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

import javax.persistence.*;
import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;

@Entity
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Patient {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long uid;
    @Email(message = "Email must be valid")
    @NotBlank(message = "Email is required")
    // uniqueness when adding new row needs to be checked in contronller
    @Column(unique = true)
    private String email;
    private String firstname;
    private String lastname;

    private String dob;
    private String password;
    private String mobilenumber;
    private String medicalhistory;

    public String toString() {
        return " email= " + email + " firstName= " + firstname + " lastName= " +
                lastname +
                " dob= " + dob + " password= " + password + " mobileNumber= " + mobilenumber
                + " medicalHistory= "
                + medicalhistory;
    }

}

File: Frontend/lib/patient.dart
class Patient {
  String dob;
  String email;
  String firstname;
  String lastname;
  String mobilenumber;
  String password;
  String medicalhistory;

  Patient(this.dob, this.email, this.firstname, this.lastname,
      this.mobilenumber, this.password, this.medicalhistory);

  Map toJson() => {
        'dob': dob,
        'email': email,
        'firstname': firstname,
        'lastname': lastname,
        'mobilenumber': mobilenumber,
        'password': password,
        'certificate': medicalhistory
      };

  factory Patient.fromJson(dynamic json) {
    return Patient(
        json['dob'] as String,
        json['email'] as String,
        json['firstname'] as String,
        json['lastname'] as String,
        json['mobilenumber'] as String,
        json['password'] as String,
        json['certificate'] as String);
  }
}
